src/models/adapters.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "../generators.h"
+#include "model.h"
+
+namespace Generators {
+
+Adapter::Adapter(const char* adapter_file_path, Ort::Allocator* allocator)
+    : adapter_{OrtLoraAdapter::Create(fs::path(adapter_file_path).c_str(), *allocator)} {}
```

Comment:
What does fs::path(adapter_file_path).c_str() do vs just passing adapter_file_path?

Reply:
It manages the os specific differences with regards to the path file (wide string vs regular string).

On Windows, the ort api expects const w_char* and on linux it expects const char*. The fs::path utility inside ort-genai manages that.